_lectures/lect13.md

```diff
@@ -0,0 +1,87 @@
+---
+num: "Lecture 13"
+desc: "Sorting Algorithms"
+ready: true
+lecture_date: 2024-05-07 15:30:00 -0700
+---
+
+# In-class activity: sorting algorithm excerises
+[54, 26, 93, 17, 77, 31, 44, 55, 20]
+
+Bubble sort:
+- Pass 1: [26, 54, 17, 77, 31, 44, 55, 20, 93]
+- Pass 2: [26, 17, 54, 31, 44, 55, 20, 77, 93]
+- Pass 3: [26, 17, 31, 44, 54, 20, 55, 77, 93]
+- Pass 4: ...
+  
+Selection sort:
+- [54, 26, 20, 17, 77, 31, 44, 55, 93]
+- [54, 26, 20, 17, 55, 31, 44, 77, 93]
+- ...
+
+# Bubble Sort
+```py
+def bubbleSort(alist):
+  for passnum in range(len(alist)-1, 0, -1):
+    printf("{passnum = }")
+    for i in range(passnum):
+      if i == 3:
+        break
+      if alist[i] > alist[i+1]:
+        temp = alist[i]
+        alist[i] = alist[i+1]
+        alist[i+1] = temp
+    print(alist)
+```
+
+# Insertion Sort
+```py
+def insertionSort(alist):
+  for index in range(1, len(alist)):
+    print("pass", index)
+    currentvalue = alist[index]
+    position = index
+
+    while position > 0 and alist[position-1]>currentvalue:
+      print("Comparing ", alist[position-1], currentvalue)
+      alist[position] = alist[position-1]
+      position = position-1
+
+    alist[position] = currentvalue
+    print(alist)
+```
+
+# Class examples
+```py
+class Pet:
+  def __init__(self, name, animal, rating):
+    """
+    :param str name: - name of pet
+    :param str animal: - type of pet(eg: 'snake' > 'cat' > 'dog')
+    :param str rating: - user rating ('meh' < 'ok' < 'ok' < 'nice' < 'awesome')
+    """
+    self.name = name
+    self.animal = animal
+    self.rating = rating
+
+  def __repre__(self):
```

Comment:
It's just `__repr__`

What about the Assertions and the comments about needing the parentheses?

It would also be helpful to add the second version of the `__lt__` where we added the rating to the mix.

Reply:
@ykharitonova I updated `__repr__` as well as the `__lt__` methods. The Q&A about needing parentheses in assertion statements as well as the Q&A about backslashes in return statements are both in the lect11.md notes (pull request still in review). I've also copied and pasted them in this lecture so students can find them in both places.